__tests__/app.test.js

```diff
@@ -168,4 +167,94 @@ describe("api/articles", () => {
         });
     });
   });
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("Status 201: creates new comment and responds with the posted comment", () => {
+      return request(app)
+        .post("/api/articles/9/comments")
+        .send({
+          username: "rogersop",
+          body: "I love articles like this.",
+        })
+        .expect(201)
+        .then(({ body }) => {
+          expect(body).toMatchObject({
+            body: "I love articles like this.",
+            votes: 0,
+            author: "rogersop",
+            article_id: 9,
+            comment_id: 19,
+          });
+          expect(body).toHaveProperty("created_at", expect.any(String));
+        });
+    });
+    test("Status 404: Responds with 'Not found' message, for valid but non-existant id.", () => {
+      return request(app)
+        .post("/api/articles/100/comments")
+        .send({
+          username: "rogersop",
+          body: "I love articles like this.",
+        })
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe("Not found");
+        });
+    });
+    test("Status 400: Responds with 'Bad request' message, for invalid id", () => {
+      return request(app)
+        .post("/api/articles/banana/comments")
+        .send({
+          username: "rogersop",
+          body: "I love articles like this.",
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request");
+        });
+    });
+    test("Status 400: Responds with 'Bad request' message, for request missing required fields", () => {
+      return request(app)
+        .post("/api/articles/9/comments")
+        .send({})
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request");
+        });
+    });
+    test("Status 400: Responds with 'Bad request' message, for request missing username value", () => {
+      return request(app)
+        .post("/api/articles/9/comments")
+        .send({
+          username: "",
+          body: "I love articles like this.",
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request");
+        });
+    });
+    test("Status 400: Responds with 'Bad request' message, for request missing body values", () => {
+      return request(app)
+        .post("/api/articles/9/comments")
+        .send({
+          username: "rogersop",
+          body: "",
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Missing comment body");
+        });
+    });
+    test("Status 400: Responds with 'Bad request' message, for invalid username.", () => {
```

Comment:
I would suggest this should be a 404, as the username is technically valid it just hasn't been found on the users table and therefore has violated the foreign key restraint.